lib/worldwide/address.rb

```diff
@@ -148,7 +148,14 @@ def valid?
     end
 
     def concatenate_address1
-      additional_fields = region.combined_address_format["address1"] || []
+      detected_scripts = Worldwide::Scripts.identify(text: street_name).map(&:to_s)
+      scripts = region.combined_address_format.except("default").keys
+      script = "default"
+      if detected_scripts.length == 1 && scripts.include?(detected_scripts.first)
+        script = detected_scripts.first
+      end
+
+      additional_fields = region.combined_address_format.dig(script, "address1") || []
```

Comment:
Logic for detecting scripts, roughly the same for all address string manipulation functions.